tests/unit/dataactcore/test_job_queue.py

```diff
@@ -1,55 +1,76 @@
+from collections import OrderedDict
+import csv
 import os
+from unittest.mock import Mock
 
+from pytest import raises
 
-@pytest.fixture()
-def mock_file_d_path(tmpdir):
-    original = CONFIG_BROKER['d_file_storage_path']
-    CONFIG_BROKER['d_file_storage_path'] = str(tmpdir) + os.path.sep
-    yield tmpdir
-    CONFIG_BROKER['d_file_storage_path'] = original
+from dataactcore.utils import jobQueue
+from dataactcore.utils.responseException import ResponseException
 
 
-def test_generate_d_file_success(monkeypatch, mock_file_d_path):
+def test_generate_d_file_success(monkeypatch, mock_broker_config_paths):
     """ Test successful generation of D1 and D2 files """
     local_file_name = "12345_test_file.csv"
-    file_path = mock_file_d_path.join(local_file_name)
+    file_path = mock_broker_config_paths['d_file_storage_path'].join(
+        local_file_name)
 
     assert not os.path.isfile(str(file_path))
 
     file_path.write("test")
 
     result_xml = "<results>test_file.csv</results>"
 
-    jq = JobQueue()
-    monkeypatch.setattr(jq, 'update_d_file_status', Mock())
-    monkeypatch.setattr(jq, 'download_file', Mock())
-    monkeypatch.setattr(jq, 'get_xml_response_content',
+    monkeypatch.setattr(jobQueue, 'download_file', Mock())
+    monkeypatch.setattr(jobQueue, 'get_xml_response_content',
                         Mock(return_value=result_xml))
-
-    jq.generate_d_file(Mock(), 1, 1, Mock(), local_file_name, True)
+    jobQueue.generate_d_file(Mock(), 1, 1, Mock(), local_file_name, True)
 
     assert os.path.isfile(str(file_path))
 
 
-def test_generate_d_file_failure(monkeypatch, mock_file_d_path):
+def test_generate_d_file_failure(monkeypatch, mock_broker_config_paths):
     """ Test unsuccessful generation of D1 and D2 files """
     local_file_name = "12345_test_file.csv"
-    file_path = mock_file_d_path.join(local_file_name)
+    file_path = mock_broker_config_paths['d_file_storage_path'].join(
+        local_file_name)
 
     assert not os.path.isfile(str(file_path))
 
-    result_xml = ""
+    monkeypatch.setattr(jobQueue, 'download_file', Mock())
+    monkeypatch.setattr(jobQueue, 'get_xml_response_content',
+                        Mock(return_value=''))
+    with raises(ResponseException):
+        jobQueue.generate_d_file(Mock(), 1, 1, Mock(), local_file_name, True)
+
 
-    jq = JobQueue()
-    monkeypatch.setattr(jq, 'update_d_file_status', Mock())
-    monkeypatch.setattr(jq, 'get_xml_response_content',
-                        Mock(return_value=result_xml))
+def read_f_file_rows(suffix, file_path):
+    jobQueue.generate_f_file(1, 1, 1, Mock(), suffix, is_local=True)
 
-    with raises(ResponseException):
-        jq.generate_d_file(Mock(), 1, 1, Mock(), local_file_name, True)
+    assert os.path.isfile(file_path)
+
+    with open(file_path) as f:
+        return [row for row in csv.reader(f)]
+
+
+def test_generate_f_file(monkeypatch, mock_broker_config_paths):
```

Comment:
There's not really a "fail" state, here -- we'd just not get any results. This tests focuses on how the CSVs are generated rather than the underlying queries (which are tested elsewhere)